Pod/Classes/RZIntrinsicContentSizeTextView.m

```diff
@@ -0,0 +1,414 @@
+@property (weak, nonatomic, readonly) NSLayoutConstraint *rz_maxHeightConstraint;
+
+// Size Helpers
+- (CGFloat)intrinsicContentHeight;
+
+// Delegate helpers
+- (UIView *)layoutView;
+- (BOOL)shouldAnimateSizeChange;
+
+// Scroll Helpers
+- (BOOL)shouldScrollToBottom;
+- (CGPoint)bottomOffset;
+
+@end
+
+@implementation RZIntrinsicContentSizeTextView
+
+@synthesize rz_minHeightConstraint = _rz_minHeightConstraint;
+@synthesize rz_maxHeightConstraint = _rz_maxHeightConstraint;
+
+#pragma mark - View Life Cycle
+
+- (id)initWithFrame:(CGRect)frame textContainer:(NSTextContainer *)textContainer
+{
+    self = [super initWithFrame:frame textContainer:textContainer];
+    if ( self ) {
+        [self configureIntrinsicTextView];
+    }
+    return self;
+}
+
+- (id)initWithCoder:(NSCoder *)aDecoder
+{
+    self = [super initWithCoder:aDecoder];
+    if ( self ) {
+        [self configureIntrinsicTextView];
+    }
+    return self;
+}
+
+- (void)dealloc
+{
+    [[NSNotificationCenter defaultCenter] removeObserver:self
+                                                    name:UITextViewTextDidChangeNotification
+                                                  object:nil];
+}
+
+- (void)prepareForInterfaceBuilder
+{
+    [super prepareForInterfaceBuilder];
+    [self configurePlaceholder];
+    [self adjustHeightIfNeededAnimated:NO];
+}
+
+#pragma mark - Layout
+
+- (void)layoutSubviews
+{
+    [super layoutSubviews];
+
+    [self adjustHeightIfNeededAnimated:self.shouldAnimateSizeChange];
+}
+
+- (CGSize)intrinsicContentSize
+{
+    return CGSizeMake(CGRectGetWidth(self.bounds), self.intrinsicContentHeight);
+}
+
+#pragma mark - Setters
+
+// custom setters
+- (void)setHeightPriority:(CGFloat)heightPriority
+{
+    if ( _heightPriority != heightPriority ) {
+        _heightPriority = heightPriority;
+
+        [self rz_replacePinnedHeightConstraintWithNewPinnedHeight:self.intrinsicContentHeight priority:self.heightPriority];
+    }
+}
+- (void)setPlaceholder:(NSString *)placeholder
+{
+    if ( ![_placeholder isEqualToString:placeholder] ) {
+        _placeholder = placeholder.copy;
+        self.placeholderLabel.text = _placeholder;
+    }
+}
+
+- (void)setAttributedPlaceholder:(NSAttributedString *)attributedPlaceholder
+{
+    if ( ![_attributedPlaceholder isEqualToAttributedString:attributedPlaceholder] ) {
+        _attributedPlaceholder = attributedPlaceholder.copy;
+        self.placeholderLabel.attributedText = _attributedPlaceholder;
+    }
+}
+
+- (void)setPlaceholderTextColor:(UIColor *)placeholderTextColor
+{
+    if ( _placeholderTextColor != placeholderTextColor ) {
+        _placeholderTextColor = placeholderTextColor;
+        self.placeholderLabel.textColor = _placeholderTextColor;
+    }
+}
+
+#pragma mark - Super Setter Overrides
+
+- (void)setTextContainerInset:(UIEdgeInsets)textContainerInset
+{
+    [super setTextContainerInset:textContainerInset];
+    [self adjustPlaceholderPosition];
+    [self scrollToBottom];
+}
```

Comment:
is it always going to the case that the once setting the inset scrollToBottom should be called?